### ISSUE TITLE
Fix FP S6594 (`sonar-prefer-regexp-exec`): Ignore match count use case

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4532]
The rule should not suggest using `Regex.exec` if the returned value of `String.match` is eventually used to get the number of matches. It only involves getting the `.length` of the value, while a loop is needed in the case of `Regex.exec`.

```js
const regex = new RegExp(pattern);
const matches = 'foo'.match(regex); // Currently false positive, should be compliant

/* ... */

matches.length;

/* or */

matches?.length;
```
